### ISSUE TITLE
Test delayed data streaming

### ISSUE THREAD

**Issue by tir38** [andrzejchm/RESTMock#109]
I have a PUT request that slowly streams bytes to a server. Internal to this call I track total bytes uploaded. I now want to write a test asserting that X bytes were uploaded. To do this I mostly need to keep RESTMock from immediately returning 200. It should let me write some bytes and only later return a response code. IIUC from other issues, `.delay()` operator only delays the response body, not the response code. Is there any way to change this?

**Comment by andrzejchm** [comment]
0.4.1 version adds a `delayHeaders(...)` method which does delay the http response codes

**Comment by tir38** [comment]
Tested and works great. Thanks!

**Comment by tir38** [comment]
Sorry for the late return but I figured out part of the reason this was not working. We also have a `"Expect: 100-continue"` header on that request. So the bug was not that restmock was delaying the response header. It was that MockWebServer was delaying the initial `100` response. When I said back in Dec that things were working, they only worked if I removed that header. Seems like this is a known issue in MockWebServer https://github.com/square/okhttp/issues/4698

Either way thanks for adding this part.
